libs/resolver/loader/module_loader.rs

```diff
@@ -221,6 +221,7 @@ impl<TSys: ModuleLoaderSys> ModuleLoader<TSys> {
         // If we loaded a JSON file, but the "requested_module_type" (that is computed from
         // import attributes) is not JSON we need to fail.
         if loaded_module.media_type == MediaType::Json
+          && !self.in_npm_pkg_checker.in_npm_package(specifier)
           && !matches!(requested_module_type, RequestedModuleType::Json)
```

Comment:
Maybe swap these? The `in_npm_package` check is more expensive.

That said, we should probably make this work even when using cjs outside npm packages and also we still want to surface this error for ESM in npm packages as Nathan mentioned.